### PR TITLE
Added RGBMATRIX_INDICATORS_OVERRIDE_RGB_OFF config macro

### PR DESCRIPTION
**Pull request by ach5948**
## Description

Added config macro to enable rgb matrix indicator leds to stay on when rgb matrix is disabled

docs/config_options.md: Added documentation for `RGBMATRIX_INDICATORS_OVERRIDE_RGB_OFF`
docs/feature_rgb_matrix.md: Added documentation for `RGBMATRIX_INDICATORS_OVERRIDE_RGB_OFF`
quantum/rgb_matrix/rgb_matrix.c: If `RGBMATRIX_INDICATORS_OVERRIDE_RGB_OFF` is defined, call `rgb_matrix_indicators` and `rgb_matrix_indicators_advanced` even if RGB matrix is turned off. Continually flush for RGB_MATRIX_NONE effect if `RGBMATRIX_INDICATORS_OVERRIDE_RGB_OFF` is defined

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
why not use the set flags function?  

eg: 
`rgb_matrix_set_flags(LED_FLAG_NONE)`

**Comment by drashna** [comment]
the flags control what LEDs are rendered (as well as some effects).   You can filter by flag, too. 

Two examples of this are the helper function I use for layer/layout indication, and how the ZSA boards handle this. 

https://github.com/qmk/qmk_firmware/blob/master/users/drashna/rgb_matrix_stuff.c#L42

https://github.com/qmk/qmk_firmware/blob/c9c2916829eccab2222dad99c7f599988789ec2f/keyboards/moonlander/moonlander.c#L444-L460

By setting the flag to "none", it disables rendering of animations, but stuff like indicators functions still work.  The caveat is that you do have to "clean up" after your LEDs.  Eg, if you turn an LED off, you have to explicitly call `rgb_matrix_set_color(i, 0, 0, 0)`.

**Review by fauxpark — COMMENTED** [review]
Should be done for LED Matrix too.

**Comment by ach5948** [comment]
Sorry about the radio silence on this. I haven't time to look at it recently. I'm planning on coming back to it soon

File: quantum/rgb_matrix/rgb_matrix.c [hunk 1]
```diff
@@ -425,7 +427,10 @@ void rgb_matrix_task(void) {
             break;
         case RENDERING:
             rgb_task_render(effect);
-            if (effect) {
+#ifndef RGBMATRIX_INDICATORS_OVERRIDE_RGB_OFF
+            if (effect)
+#endif
+            {
```
Comment (blueyed) [inline quantum/rgb_matrix/rgb_matrix.c:433]: I assume this is meant to still call it when using `RGB_MATRIX_NONE`.
Note that using `effect` there was changed in: https://github.com/qmk/qmk_firmware/commit/12568fb5a9167d29a52f79c739f11830bde3e4be (#11308), i.e. previously it would work as with the new `#define`.

File: quantum/rgb_matrix/rgb_matrix.c [hunk 2]
```diff
@@ -387,10 +387,12 @@ static void rgb_task_render(uint8_t effect) {
     // next task
     if (!rendering) {
         rgb_task_state = FLUSHING;
+#ifndef RGBMATRIX_INDICATORS_OVERRIDE_RGB_OFF
         if (!rgb_effect_params.init && effect == RGB_MATRIX_NONE) {
             // We only need to flush once if we are RGB_MATRIX_NONE
             rgb_task_state = SYNCING;
         }
+#endif
```
Comment (blueyed) [inline quantum/rgb_matrix/rgb_matrix.c:395]: Why is this needed?

**Review by blueyed — COMMENTED** [review]
I also wondered why `RGB_MATRIX_NONE` would not be used more (e.g. `RGB_MOD` does not select it etc), when looking briefly into removing/avoiding overhead with the RGB matrix feature (`rgb_matrix_task` in particular), when using custom indicators only (with no effect).

Some recent PR related to this: https://github.com/qmk/qmk_firmware/pull/14997 (https://github.com/qmk/qmk_firmware/commit/a63c2c5f509e98dc58ea876f525896aa717a0d6e).
